File: code/python/Deck.py
from Card import Card
import random

class Deck:
    def makeDeck(self):
        self.pile=[]
        cards=[x for x in range(14)[1:]]
        suits=['s','h','c','d']
        for s in suits:
            for c in cards:
                tempCard=Card(val=c,suit=s)
                self.pile.append(tempCard) 
    
    def shuffleDeck(self):
        random.shuffle(self.pile)
    
    #if deck is empty will return a card valled "InvalidCardName_InvalideCardSuit"
    def deal(self)->Card:
        if len(self.pile)>0:
            return self.pile.pop(0)
        else:
            emptyDeckCard=Card(val=-1, suit="noSuit")
            return emptyDeckCard

    def __init__(self):
        #self.pile=list of cards 
        #self.makeDeck() will instantiate a deck in order (ace-king)(s,h,c,d)
        #self.shuffle() will randomize a deck
        #self.deal() pop one off self.deck and return it
        self.makeDeck()

File: code/c++/Deck.cpp
#include "Deck.h"
#include "Card.cpp"
#include <string>
#include <algorithm>
#include <vector>
Card Deck::deal(){
    if(index>0){
        Card temp=pile.at(index);
        index--;
        pile.pop_back(); 
        return temp;
    }
}
void Deck::makeDeck(){
    index=51;
    char suits[]={'S', 'H', 'C', 'D'};
    for(int s=0;s<4;s++){
        for(int v=1;v<14;v++){
            pile.push_back(Card(v,suits[s]));
        }
    }
}
void Deck::shuffleDeck(){
    std::random_shuffle(pile.begin(), pile.end());
}
Deck::Deck(){
    index=51;
    makeDeck();
}
Deck::~Deck(){
    pile.clear();
}